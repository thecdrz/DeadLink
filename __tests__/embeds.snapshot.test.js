const { dashboardEmbed, legendField } = require('../lib/embeds');

describe('embeds snapshot', () => {
  test('dashboardEmbed with legend remains stable', () => {
    const connFmt = { emoji: 'ðŸŸ¢', text: 'Online' };
    const discordFmt = { emoji: 'ðŸŸ¢', text: 'Online' };
    const embed = dashboardEmbed({ connFmt, discordFmt, modeMsg: 'Dev', version: '9.9.9' });
    // Ensure legend is appended (as the UI uses)
    embed.fields = embed.fields || [];
    embed.fields.push(legendField());
    expect(embed).toMatchSnapshot();
  });
});
