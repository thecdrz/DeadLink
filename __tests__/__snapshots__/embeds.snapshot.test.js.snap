// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`embeds snapshot dashboardEmbed with legend remains stable 1`] = `
{
  "color": 7506394,
  "description": "🟢 **Server Status**: Online   •   🟢 **Discord**: Online
🛠️ **Mode**: Dev

Welcome to the interactive server control panel! Use the buttons below to quickly access server information and analytics.

🎯 **Activity** - Get detailed player activity reports
📊 **Trends** - View player count analytics and trends
👥 **Players** - See current online players
⏰ **Time** - Use /time to check current game time
ℹ️ **Info** - Server version and details",
  "fields": [
    {
      "inline": false,
      "name": "Legend",
      "value": "⚔️ Kills · ☠️ Deaths · 📏 Distance · 🔥 Streak · ❤️ Health · 🧪 Dev",
    },
  ],
  "footer": {
    "text": "DeadLink v9.9.9",
  },
  "title": "🎮 7 Days to Die Server Dashboard",
}
`;
